RUFAS/routines/animal/animal_management.py

```diff
@@ -331,17 +330,66 @@ def calc_nutrient_rqmts(self, feed, temp):
         for cow in self.cows:
             cow.set_nutrient_rqmts()
 
-    def fully_update_id_pen(self):
+    def fully_update_animal_to_pen_id_map(self):
         """
-        Updates the entire id_pen dictionary so that each animal's ID is
+        Updates the entire animal_to_pen_id_map dictionary so that each animal's ID is
         associated with the pen that animal is in.
         """
-        for pen in self.all_pens:
+        for pen in self.all_pens_ids:
             animals_in_pen = pen.animals_in_pen
             for animal in animals_in_pen:
-                self.id_pen[animal.id] = pen.id
+                self.animal_to_pen_id_map[animal.id] = pen.id
+
+    def remove_animal_from_herd(self, animals_removed):
```

Comment:
```suggestion
    def remove_animals_from_herd(self, animals_removed):
```

Would this make more sense to be `remove_animals_from_herd` because its removing multiple animals?

Reply:
Thanks for catching that, good point!